pulp_container/app/authorization.py

```diff
@@ -92,32 +118,70 @@ def _split_into_encoded_groups(self, encoded_base32):
                 kid += char
         return kid
 
-    def determine_access(self, user, scope):
+    def determine_access(self):
         """
         Determine access permissions for a corresponding user.
 
         This method determines whether the user has a valid access permission or not.
-        The determination is based on role based access control. For now, the access
-        is given out to anybody because the role based access control is not implemented
-        yet.
-
-        Args:
-            user (str): A name of the user who is trying to access a registry.
-            scope (str): A requested scope.
+        The determination is based on role based access control.
 
         Returns:
             list: An intersected set of the requested and the allowed access.
 
         """
-        typ, name, actions = scope.split(":")
-        actions_list = actions.split(",")
-        permissions = {"pull"}
-        if user == "admin":
-            permissions.add("push")
-        permitted_actions = list(set(actions_list).intersection(permissions))
-        return {"type": typ, "name": name, "actions": permitted_actions}
-
-    def _generate_claim_set(self, issuer, issued_at, subject, audience, access):
+        typ, name, actions = self.scope.split(":")
+        actions = set(actions.split(","))
+
+        permitted_actions = set()
+        for action in actions:
+            has_permission = self.actions_permissions[action](name)
+            if has_permission:
+                permitted_actions.add(action)
+
+        return {"type": typ, "name": name, "actions": list(permitted_actions)}
+
+    def has_pull_permissions(self, path):
+        """
+        Check if the user has permissions to pull from the repository specified by the path.
+        """
+        if self.user.has_perm("container.view_containerdistribution"):
+            return True
+
+        try:
+            distribution = ContainerDistribution.objects.get(base_path=path)
+        except ContainerDistribution.DoesNotExist:
+            return False
+        if self.user.has_perm("container.view_containerdistribution", distribution):
```

Comment:
https://github.com/pulp/pulp_container/pull/203#discussion_r557510009
Maybe we should add a specific "pull" permission here, because "view_containerdistribution" includes reading the distributions properties.

Reply:
that is ok i think, why that would be a problem?